[Hub Generated] Review request for Microsoft.CostManagement to add version stable/2019-11-01

sashahdo opened the pull request:
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sashahdo/azure-rest-api-specs/dev-cost-management-Microsoft.CostManagement-2019-11-01...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

sashahdo committed:
added forecast api and all of it's scopes

sashahdo committed:
added alerts path and it's example files

sashahdo committed:
fixed typos in alertProperties

sashahdo committed:
fixed spacing as per prettier script

sashahdo committed:
differentiated forecastDefinition x-ms-enums

sashahdo committed:
changed forecast operationID and forecast fixed response ids to correctly reflect current state in prod.

sashahdo committed:
added new path for individual alerts

sashahdo committed:
added new path for individual alerts

sashahdo committed:
fixed incorrect response type

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

bitsforthought committed:
[Azure Cognitive Search] Various Swagger improvements (#9301)

* [Azure Cognitive Search] Add warning to Delete Index description

* [Azure Cognitive Search] Rename searchMode to SuggesterSearchMode

Trying to avoid ambiguity between generated model names in SDK code.

* [Azure Cognitive Search] Remove access-condition parameter group

The parameter group was for codegen purposes, but it turns out that clients
will use a different pattern for exposing ETag capabilities, so this is no
longer needed. However, we're keeping the If-Match and If-None-Match
parameters themselves for REST API documentation purposes.

Fixes https://github.com/Azure/azure-rest-api-specs/issues/9205

* [Azure Cognitive Search] Clarify docs for pattern analyzer & tokenizer

Fixes https://github.com/Azure/azure-rest-api-specs/issues/9145

* [Azure Cognitive Search] Document more enums

Fixes https://github.com/Azure/azure-rest-api-specs/issues/9146

* [Azure Cognitive Search] Make List Indexes "fake pageable"

This makes codegen easier for now, with the expectation that this API may
become pageable in the future.

Fixes https://github.com/Azure/azure-rest-api-specs/issues/9274

* [Azure Cognitive Search] Rename models per arch board feedback

Fixes https://github.com/Azure/azure-rest-api-specs/issues/9295

* [Azure Cognitive Search] Add names to dictionary

* [Azure Cognitive Search] Improve the description of analyzingInfixMatching

* [Azure Cognitive Search] Change prefix match example to pacify spell checker

Co-authored-by: Bruce Johnston <brjohnst@microsoft.com>

sashahdo committed:
Merge branch 'master' into dev-cost-management-Microsoft.CostManagement-2019-11-01

AutorestCI commented:
/azp run automation - sdk

sashahdo committed:
added whitespace as per prettier script

sashahdo committed:
added whitespace as per prettier script

sashahdo committed:
Merge branch 'dev-cost-management-Microsoft.CostManagement-2019-11-01' of https://github.com/sashahdo/azure-rest-api-specs into dev-cost-management-Microsoft.CostManagement-2019-11-01

AutorestCI commented:
/azp run automation - sdk

jaysterp committed:
Add Pipeline Resources for Image Transfer (#8662)

* update 2019-12-01-preview swagger for import and export pipelines

* add pipeline runs resource

* add examples for import and export pipelines

* add pipeline run examples

* fix importpipeline example responses

* update example

* remove isArchiveEnabled property so the used can explictly tell when to delete

* fix model validation in pipelineRun examples

* resolve duplicate model conflict

* run prettier

* update examples for pipeline run

Grayer123 committed:
Bug fixes for two known issues (#9272)

* Bug fixes: skip url encoding for ScopeParameter; add client name for userAssignedIdentities.

* Pretty check fix.

viananth committed:
[Azure Stack] Updates to Azs.Keyvault.Admin spec (#8974)

* Cumulative commit message:
commit 9e551f0eab4057d4c2f54c333c7aa2a1a564c125
Author: bganapa <bganapa@microsoft.com>
Date:   Tue Nov 12 11:44:22 2019 -0800

    Reset to Stackadmin2 (#7766)

* Add autorest gen info to readme.md

Commit:
regenerated all-api-versions

ChenTanyi committed:
feat: update desktop virtualization java readme (#9337)

* feat: update desktop virtualization java readme

* fix: resource java readme

yangyuan committed:
Add 'capabilities' field to Cognitive Services account properties. (#9309)

yangyuan committed:
Add private endpoint support for Cognitive Services. (#9298)

* Add private endpoint support for Cognitive Services.

* Address comments

* fix prettier

mvvsubbu committed:
Adding NoPublicIP feature back (#9306)

lwang2016 committed:
Update deploymentScripts.json (#9325)

Mark environment variable secureValue as secret

chenjianfei2017 committed:
Windowsiot (#9333)

* resourcegraph trenton

* trenton pipeline

* test windowsiot pipeline

Co-authored-by: root <root@cjf.1nfgrxx31qve3h2ogd1kz2gjvd.cx.internal.cloudapp.net>

allenjzhang commented on specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:
```diff
@@ -317,6 +317,103 @@
+          }
+        }
+      },
+      "patch": {
```
Mmm, not sure about using Patch as an action (not just for updating state to signal dismiss state). This is typically done with POST. Calling it out for ARM review.

sashahdo replied:
thanks for pointing out the enum issue. I made an update adding all missing enums.

majastrz replied:
I'm really confused about the purpose of the API. The contract looks like a legitimate PATCH operation, but you call it a "dismiss" API.

If you are planning to allow patching of other alert properties in the future, then I think it makes sense to keep PATCH as the verb but you should change all the descriptions. ("Dismiss" in examples is fine, though.)

If dismissing is the only modification that you will allow, then it should be a POST /dismiss on the alert resource.

---
In reply to: [421001556](https://github.com/Azure/azure-rest-api-specs/pull/9329#discussion_r421001556) [](ancestors = 421001556)

theelderwand replied:
@majastrz  That's great feedback. Completely agree with you on this one. It should be a /dismiss. However, this is the currently published API which customers are actively using so we are not looking to make any breaking changes to contract here. When we do rev the API version in the near future we will incorporate this.
Do let me know if you see any concerns on this.

majastrz replied:
It's fine to keep it, but you should update the description accordingly (except for examples). That change is swagger-only and has no bearing on your backend.

shail2208 committed:
Adding userassigned identities for compute (#8879)

* Added User identities for compute

* Changing the api version

* Revert "[Do not Merge] Adding user identities"

* Adding userassigned identities for compute

* running prettier

* Updating Quota examples

* Undoing Quota examples

* Updating the examples fine

* Updating Quota examples

Co-authored-by: Shail Paragbhai Shah <shaisha@microsoft.com>
Co-authored-by: Sriram Kakara <skakara@microsoft.com>

Commit:
[Storage] Add support for listing private endpoint connections (#9324)

* [Storage] Add support for listing private endpoint connections

This change adds support for listing private endpoint connections through a GET collection API for Microsoft.Storage/storageAccounts/privateEndpointConnections.

* Update StorageAccountListPrivateEndpointConnections.json

Fixed prettier check fix.

* Update storage.json

Fixed lintdiff build error

akning-ms committed:
Revert "remove shanghai assignee during 5.1 holiday" (#9354)

* Revert "remove shanghai assignee during 5.1 holiday (#9254)"

This reverts commit e4fd866ef08eb5e5a95caede3c915e27cec0772f.

* remove Allen as reviewer in May as requested

jeffj6123 committed:
service Fabric 7.1 (#9291)

* added SF 7.1

* fixed spelling

* added reference to main file

* removed unused files

* updated tag

Commit:
regenerated all-api-versions

sashahdo committed:
added all missing alert enums

sashahdo committed:
added all missing alert enums

AutorestCI commented:
/azp run automation - sdk

sashahdo committed:
fixed whitespace as per prettier script

sashahdo committed:
fixed whitespace as per prettier script

AutorestCI commented:
/azp run automation - sdk

qiaozha committed:
Change python client name (#9278)

* change python client name

* change python client name

* change title overrider

arolshan committed:
fix folder name for schema auto generation (#9303)

ChenTanyi committed:
feat: update java readme for network (#9341)

weidongxu-microsoft committed:
modelerfour need the x-ms-api-version extention to notify it that this api-version does not need special handling (#8803)

Nilambari committed:
Revert commit:-  CustomDnsServer optional param added for P2SVpnGateway resource from April release. (#9365)

* Revert "Fix json pretty errors"

This reverts commit bd5293975a73fe64d93af00debf93a87fefa29bc.

* Revert "Fix example"

This reverts commit 827e8829fa36678ba2247ff51cec40b5f8cf985c.

* Revert "Custom dns servers for Virtual Wan P2SVpnGateway"

This reverts commit a91490a45a80a370ec657a8ed7556a48d6be9d65.

johnthcall committed:
[Hub Generated] Review request for Microsoft.ResourceGraph to add version preview/2018-09-01-preview (#9216)

* feedback

* Linter

* Remove additional fields

* Revert operation changes

* Undo etag capitalization

JasonYang-MSFT committed:
Updated code owner for storage (#9338)

viananth committed:
[Azure Stack] Updates to Azs.Commerce.Admin spec for autorest config (#9033)

* Cumulative commit message:
commit 4ec980f9748c6ee9a98389fff0b3941fc8f1fc63 (azure/stackadmin)
Author: Yuxing Zhou <zyx.pulsars@gmail.com>
Date:   Wed Mar 18 15:14:57 2020 +0800

    [Azure Stack] Add scaleUnit and volumeLabel parameter to Disks_List (#8733)

    * add scaleUnit and volumeLabel to Disks_List

    * fix ref location

    * update parameter description

commit dd09a40a97fdfd2b215498a6643ce21c9a2dbd38
Author: bganapa <bganapa@microsoft.com>
Date:   Tue Mar 17 17:32:10 2020 -0700

    Update to camelCasing for resourceGroups

commit 4212dd05b892d364f7271719847c4adeb896a9e6
Author: bganapa <bganapa@microsoft.com>
Date:   Tue Mar 17 17:30:07 2020 -0700

    Update to camelCasing for resourcegroups                                                                                                                                                                                                    commit b62db8ba4e29e14935ff33e82bda029393eb8393                                                                         Merge: 08d597aa1 fbe767051                                                                                              Author: Vishnu Priya Ananthu Sundaram <viananth@microsoft.com>                                                          Date:   Thu Mar 12 11:23:30 2020 -0700
    Merge branch 'stackadmin' of https://github.com/Azure/azure-rest-api-specs into stackadmin

commit fbe767051ac6d3c324f7e4d028e7ec5639120407
Author: bganapa <bganapa@microsoft.com>
Date:   Tue Mar 10 14:06:55 2020 -0700

    Update resourceGroups casing

commit 848e5f76bc6983599cedb4a7e7cd1666ffd8a298
Author: bganapa <bganapa@microsoft.com>
Date:   Tue Mar 10 14:05:10 2020 -0700

    Update resourcegroups casing to resourceGroups

commit 591ff3c47d52bdfbe228ceadb2c8a11781630e99
Author: bganapa <bganapa@microsoft.com>
Date:   Tue Mar 10 13:58:14 2020 -0700

    Update Updates.json

commit adaabc1bd3b17d60f9ac39b4d95e7d0df60019ca
Author: bganapa <bganapa@microsoft.com>
Date:   Wed Mar 4 18:07:20 2020 -0800

    Changing casing in resourcgroups to resourceGroups

commit 72efdec25338cfbb9fea108dcfb390f4c5ff6295
Author: bganapa <bganapa@microsoft.com>
Date:   Thu Feb 27 17:26:56 2020 -0800

    Update to have $(this-folder)/ in the input files

* Add autorest config to readme.md

* Fix path reference

* Auto-generated values

Commit:
regenerated all-api-versions

changlong-liu committed:
Cli onboard desktopvirtualization (#9367)

* cli validation for desktopvirtualization

* correct readme title

Co-authored-by: Changlong Liu <changll@microsoft.com>

jerryc3 committed:
[Compute] [SIG] [BugFix] Remove ID as required because update gallery image version is failing when ID is not present (#9257)

* remove id as required

* update for july too

Co-authored-by: Jerry Chan <jerch@microsoft.com>

ArcturusZhang committed:
Fix bad readme.go.md in azuredata (#9370)

* Fix bad readme.go.md in azuredata

* Fix comments

Commit:
EventHub: added optional keyVersion property in KeyVaultProperties (#9335)

* added optional keyVersion property in KeyVaultProperties

* Keyvault properties to Array

Co-authored-by: v-ajnava <v-ajnava@microsoft.com>

KranthiKiranerusu committed:
Adding the new resource type PrivateLinkForAzureAD (#8854)

* Adding the new resource type PrivateLinkForAzureAD

* Added the example

* Updated the example

* Updated the definition for privateLinkforAzureAD

* Added privatelinkresource

* Added privatelinkresource

* Updated definitions

* Updated definitions

* Spell check

* Moved to different files

* Fixing Model validation

* added xmspageable

* Fixing example

* Added nextlinks and corrected the versions in the examples

* Added PUt/PATCH/DELETE to comply with a trackedResource

* Fixes for Model Validation + AllTenants Property

Fixed the error messages thrown by model validation as well as introduced the "allTenants" property under privatelinkpolicy

* Fixes for Prettier Check

Pushing fixes done by prettier check

Co-authored-by: Arturo Lucatero <arluca@microsoft.com>

Commit:
regenerated all-api-versions

xiaoxuqi-ms committed:
tf support secondary branch (#9378)

Co-authored-by: Xiaoxu Qi <xiaoxuqi@microsoft.com>

Commit:
attestation fix (#9077)

* fixing attestation

* undo unnecessary changes

* missing

aarongas committed:
[Hub Generated] Review request for Microsoft.Insights to add version preview/2020-02-10-preview (#8383)

* Adds base for updating Microsoft.Insights from version preview/2019-10-17-preview to version 2020-02-10-preview

* Updates readme

* Updates API version in new specs and examples

* Added WebTestResults API

* Adjusted tiles

* made changes to example file.

* Adjusted example to have better sample values

* Updated doc to produce better help file.

* corrected example file.

* added _get to operationId to match GetInOperationName requirement

* Adjustments from review

* Adjusted enum for downloadAs

* Adjusted file to make more readable.

* Added better descriptions to the file format enum.

* Removed partition ID from the example.

* Updated description

* Changes made based on Igors review.

* removed dashes from example and removed _Get from operationId

* Fixes typeo

* adjusted with prettier

* adjusted json

* adjusted to a put call and added operationslist as per code review from arm team.

* corrected to post

* added operatinsList.json example file

* adjusted file name.

* adjusted example file.

* adjusted case of microsoft.insights

* minor fix

* ran prettier-fix

* changes per code review.

* Fixed case in example file.

* removed nextLink from Operations

Commit:
regenerated all-api-versions

zfchen95 committed:
Fix the format of shareUsageBytes to show larger file share size (#9315)

MoustafaEid committed:
updated modelAsString to true (#9327)

NateLehman committed:
[Hub Generated] Review request for Microsoft.DigitalTwins to add version preview/2020-03-01-preview (#9304)

* init commit

* updates

* re-add sku as future use

* make sku writeOnly so it is not expected in return

* correction

* test

* fixup

Commit:
[AzureSearch] Add more examples on IP-restriction and private endpoint + fixes (#9381)

* update iprules and add more samples for create and update;fix incorrect sample for private endpoint

* verbose change

* typo fix

* wording fix

* fix model validation

ariklin committed:
On boarding Solutions to latest stable API version (#8630)

* Copy solutions files from 2015-06-01-preview

* update API version

* update readme

* Add required to ExternalSecuritySolution

* fix comment

* fix conflict

* fic conflict2

* fix conflict3

* fix conflict4

Co-authored-by: Arik Riklin <arik.riklin@microsoft.com>

Commit:
regenerated all-api-versions

yangyuan committed:
Revert "Add 'capabilities' field to Cognitive Services account properties. (#9309)" (#9360)

caod committed:
Add resource type sqlServerInstances (#9197)

* Add resource type sqlServerInstances

* add to resource provider - Microsoft.AzureData

* exactly same REST API definitions as existing resource types of the
  same resource provider

* Add property description and mark required properties
  # update examples to include required properties

* set createTime and updateTime property to readonly

  update CreateOrUpdateSqlServerInstance.json sample to not include readonly
  properties in its request.

Co-authored-by: Dong Cao <dongcao@microsoft.com>

zachraMSFT committed:
Remove quotes from data queries in RI apis (#9323)

TinaHu1 committed:
[DataFactory]Store settings should be added for Delete and GetMetadata (#9208)

* [DataFactory]Store settings should be added for Delete and GetMetadata Activity

* Add format settings for GetMetadata

alexaloni committed:
[Hub Generated] Review request for Microsoft.ChangeAnalysis to add version preview/2020-04-01-preview (#8981)

* New Readme Config File

* New Go Language Readme Config File

* New Typescript Language Readme Config File

* New Python Language Readme Config File

* New C# Language Readme Config File

* New Ruby Language Readme Config File

* New Swagger Spec File

* New Swagger Example Spec File

* add swagger spec second time

* trying to fix violations #1

* adding descriptions and schema tag for default error

* removing required location element from proxy resource.

* adding prettifier fixed

* update 3

* update 4

* update 5

* update 6

* update 7

* update 8

* update 9

* update 10

* update 11

* update 12

* update 13

* update 14

* update 15

* update 16

* update 17

* update 18

* Adds suppression to readme

* updating comments of few models

* changing description of few elements

* prettier  run:

Co-authored-by: Alex Dubinkov <alexdu@microsoft.com>

Commit:
regenerated all-api-versions

ArcturusZhang committed:
Fix readme.go.md for some sql service (#9348)

ArcturusZhang committed:
Update network (#9349)

ArcturusZhang committed:
Add multiapi section (#9347)

shurd committed:
Dataplane swagger for tables (#8151)

* adding swagger for tables dataplane

* updating produces/consumes, adding format parameter and adding DataServiceVersion header

* update swagger so that query parameters do not appear in x-ms-paths

* add new custom words, fix incorrect api version

* making suggested changes to continuation header, response status codes

* Updates to tables spec

* Fixed indentation

* Removed storage references

* Grammar

* Improved match condition description

* Remove format statements

* Removed excess tags

* Fix extra commas

* Added missing example

* Updated Delete example

* Run prettier

* Removed duplicate API version

* Fixed insert response

* Removed 2018 readme tag

* Remove storage name

* Remove multi-api reference

Co-authored-by: Sam Hurd <sahurd@microsoft.com>
Co-authored-by: annatisch <antisch@microsoft.com>

Commit:
regenerated all-api-versions

datr93 committed:
Datr m (#9332)

* update readme docs

* add nullables

* add autogen

* location required

* add require to all

* add requires

* prettier

* track resoruce ref and require for properties

* remove #

* fix # resoruce

* pr comments

* fix ruby

* go fix

* azure_mgmt_desktop_virtualization

* rekickoff tests

Commit:
regenerated all-api-versions

johnstairs committed:
Allow empty access policies array in HealthcareApis (#9326)

* Allow empty access polies array in HealthcareApis.

* accessPolicies property can be omitted

fore5fire committed:
[CDN] Add API version 2020-04-15 (#9242)

Commit:
regenerated all-api-versions

navysingla committed:
Update readme.md (#9384)

* Update readme.md

Adding stable version to not break existing APIs

* Added examples

* fix example

* prettier fix

Commit:
regenerated all-api-versions

Commit:
[MixedReality] Version Iteration for new Releases (#9351)

* Version Iteration for new Releases (#8848)

* Version Iteration for new Releases

There are going to be 2 significant releases:

1. Public Preview of Remote Rendering in April 6th
2. GA of Spatial Anchors in May

* Referring new swagger files in readme.md

* Specifying default package version as 2020-05-01

* Fix MV

* Fix MV (#9397)

Fix MV in Patch examples

Co-authored-by: Xiangyu Luo <xiangyul@microsoft.com>
Co-authored-by: Lei Ni <leni@microsoft.com>

Commit:
fix network (#9369)

* fix network

* duplicated publicipprefix

* fixed example in 2018-01-01

* final updates

* missing examples

Commit:
regenerated all-api-versions

Commit:
SQLDB - update LicenseType description (#9001)

HimanshuChhabra committed:
SDK support for Table and Queue Storage Services (#9259)

* Adding table spec, without examples

* Adding queue swagger, also updating comments for table

* Updated the models to include ARM fields, completed, no examples

* Added list table and queue services spec

* Adding table CRUD examples

* Adding Queue CRUD examples

* Validation complete, TablName changed to camelCase

* Modified comment

* Spell correction

* changing order in readme.md

* Changed the object name from Queue to StorageQueue, as sdk code gen throws name conflict error with Collections.Queue

* Resolve prettier check errors

* Change URI param to be camel case

* updated default error response schema

* Change Default error response model schema

* Added x-ms-nextLink

Commit:
regenerated all-api-versions

adewaleo committed:
[ACR] add regionEndpointEnabled to replication properties and replicationUpdateParameters (#8840)

* add regionendpointenabled to replication properties and replicationupdateparameters

* fix property placement error

* Update spec. with update properties

* Update descriptions

Co-authored-by: Tosin Adewale <oluwatosin.adewale@microsoft.com>

ruowan committed:
update triage label (#9396)

changlong-liu committed:
validate blockchain cli codegen (#9089)

* validate blockchain cli codegen

* use strong password with length>=12

Co-authored-by: Changlong Liu <changll@microsoft.com>

ChenTanyi committed:
feat: add subscription 2019 10 01 preview (#9388)

elabicha committed:
[Hub Generated] Review request for Microsoft.CostManagement to add version stable/2019-11-01 (#9288)

* adding the views api to the 2019-11-01 version

* Fix api version value

* adding private view list example document

* Fix enum conflict

* fix Granularity enum type

ArcturusZhang committed:
Add api version of go SDK for imagebuilder (#9408)

Commit:
Update PR template (#9371)

* Update PR template

* Add FAQ link

sashahdo committed:
Merge branch 'dev-cost-management-Microsoft.CostManagement-2019-11-01' of https://github.com/sashahdo/azure-rest-api-specs into dev-cost-management-Microsoft.CostManagement-2019-11-01

AutorestCI commented:
/azp run automation - sdk

sashahdo commented:
rebased with upstream master to resolve merge conflicts

sashahdo committed:
fixed incorrect alertsResponse

AutorestCI commented:
/azp run automation - sdk

allenjzhang commented:
@RodgeFu , Can you take a look CLI automation failure? Thanks!

sashahdo commented:
this pr has gotten quite ugly so I'm going to go ahead and abandon it and create a new one.
